controller/volume_controller.go

```diff
@@ -1363,14 +1312,16 @@ func (vc *VolumeController) getReplenishReplicasCount(v *longhorn.Volume, rs map
 			usableCount++
 		}
 	}
-
+	if v.Spec.NumberOfReplicas < usableCount {
+		return 0
+	}
 	return v.Spec.NumberOfReplicas - usableCount
 }
 
 func (vc *VolumeController) hasEngineStatusSynced(e *longhorn.Engine, rs map[string]*longhorn.Replica) bool {
 	healthyCount := 0
 	for _, r := range rs {
-		if r.Spec.FailedAt == "" {
+		if r.Spec.FailedAt == "" && r.Spec.NodeID != "" {
```

Comment:
We should change the name `healthyCount` since it's not really a healthy count. It's more like a `connectedReplicaCount`.

Reply:
Thank you! Fixed